### PR TITLE
tools: include hash in the changelog for tracing

### PR DESCRIPTION
**Pull request by Meulengracht**

### CODE REVIEW

File: tools/generate-changelog.py [hunk 1]
```diff
@@ -181,7 +186,10 @@ def main():
     new_manifest = args.new
     docs_dir = args.docs
 
-    changes = '[ Changes in primed packages ]\n\n'
+    # add a header that helps us audit where the current build is
+    # sourced from.
+    changes = f"Built from commit: https://github.com/canonical/core-base/tree/{git_commit_hash()}\n\n"
```
Comment (Meulengracht) [inline tools/generate-changelog.py:191]: Should we hardcode this URL or should we try to do some kind of magic with the ouput of `git remote get-url origin` to autogenerate it?

**Review by alfonsosanchezbeato — APPROVED** [review]
Thank you! I have some comments

**Review by alfonsosanchezbeato — CHANGES_REQUESTED** [review]
Sorry, I approved when I did not meant too

File: tools/generate-changelog.py [hunk 2]
```diff
@@ -175,13 +175,17 @@ def main():
...
 
     old_manifest = args.old
     new_manifest = args.new
     docs_dir = args.docs
 
-    changes = '[ Changes in primed packages ]\n\n'
+    # add a header that helps us audit where the current build is
+    # sourced from.
+    changes = f"Built from commit {args.permalink}\n\n"
+    changes += '[ Changes in primed packages ]\n\n'
```
[the remark this answers is not in the document]
Comment (Meulengracht) [reply tools/generate-changelog.py:202]: https://paste.ubuntu.com/p/p94sfmZwDS/ this is how it looks now - I think the indentation is not needed tbh

**Pull request merged**